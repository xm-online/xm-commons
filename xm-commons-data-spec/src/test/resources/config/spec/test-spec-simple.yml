---
test-items:
  - key: store/STORE-INFO
    inputSpec: |
      {
      "$schema": "http://json-schema.org/draft-07/schema#",
      "type": "object",
      "properties": {
        "name": { "type": "string" },
        "location": {
          "$ref": "#/xmDefinition/address"
        }
      },
      "required": ["name"]
      }
    inputForm: |
      {
        "form": [
          {
            "$ref": "#/xmForm/itemsByStoreForm"
          },
          {
            "$ref": "#/xmForm/datesForm"
          }
        ],
        "app": {
          "hideNameAndDescription": true
        }
      }
  - key: store/GET-EMPLOYEES-AGE
    inputSpec: |
      {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "type": "object",
        "properties": {
          "employeesInfo": {
            "type": "array",
            "items": {
              "type": "object",
              "properties": {
                "employeeInfo": {
                  "$ref": "#/xmDefinition/employeeInfo"
                }
              }
            }
          },
          "location": {
            "$ref": "#/xmDefinition/address"
          }
        },
        "required": ["employeesInfo"]
      }
    inputForm: |
      {
        "form": [
          {
            "$ref": "#/xmForm/itemByCategoryForm"
          },
          { "key": "role", "title": { "en": "Role" }, "type": "string", "readonly": true, "condition": {"functionBody": "{ return $.testValues; }"} },
          { "type": "submit", "title": "Submit", "condition": {"functionBody": "{return $.principal.hasPrivilegesInline(['EMPLOYEE.UPDATE.FIELDS']);}"}}
        ],
        "app": {
          "hideNameAndDescription": true
        }
      }
definitions:
  - key: address
    ref: test-spec/json-definitions/address.json
  - key: employeeInfo
    value: |-
      {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "type": "object",
        "properties": {
          "storeName": {
            "type": "string"
          },
          "info": {
            "$ref": "#/xmDefinition/userInfo"
          }
        }
      }
  - key: userInfo
    value: |-
      {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "type": "object",
        "properties": {
          "username": {
            "type": "string"
          },
          "phoneNumber": {
            "type": "string"
          },
          "age": {
            "type": "number"
          }
        }
      }
forms:
  - key: itemsByStoreForm
    value: |-
      {
        "key": "storeName",
        "title": "Enter store name",
        "type": "rest-select",
        "labelField": "name",
        "valueField": "id"
      }
  - key: itemByCategoryForm
    value: |-
      {
        "form": [
          {
            "$ref": "#/xmForm/itemsByStoreForm"
          },
          {
            "key": "name"
          }
        ],
        "app": {
          "hideNameAndDescription": true,
          "formHeaderTitle": {
            "en": "Add new"
          }
        }
      }
  - key: datesForm
    ref: test-spec/json-forms/datesForm.json
