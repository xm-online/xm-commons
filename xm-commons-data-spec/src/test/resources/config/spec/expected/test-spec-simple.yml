---
test-items:
    - key: store/STORE-INFO
      inputSpec: |
          {
            "$schema": "http://json-schema.org/draft-07/schema#",
            "type": "object",
            "properties": {
              "name": {
                "type": "string"
              },
              "location": {
                "$ref": "#/xmDefinition/address"
              }
            },
            "required": ["name"],
            "xmDefinition": {
              "address": {
                "type": "object",
                "properties": {
                  "address": {
                    "type": "string"
                  },
                  "geoAddress": {
                    "$ref": "#/xmDefinition/geoAddress"
                  },
                  "formattedAddress": {
                    "type": "string"
                  }
                }
              }
            }
          }
      inputForm: |
          {
            "form": [
              {
                "key": "storeName",
                "title": "Enter store name",
                "type": "rest-select",
                "labelField": "name",
                "valueField": "id"
              },
              {
                "type": "flex",
                "flex-flow": "row wrap",
                "fxLayoutAlign": "space-between center",
                "fxLayout": "row wrap",
                "justify-content": "space-between",
                "items": [
                  {
                    "key": "codeExpirationTime",
                    "title": {
                      "en": "Verification Code Expiration Time",
                      "uk": "Час дії коду підтвердження"
                    },
                    "type": "datetime-picker",
                    "formatString": "DD-MM-YYYY HH:mm",
                    "readonly": true,
                    "flex": "0 0 32%"
                  },
                  {
                    "key": "acceptanceTimeOfPrivacyPolicy",
                    "title": {
                      "en": "Private policy accepted at",
                      "uk": "Політика конфіденційності прийнята о"
                    },
                    "type": "datetime-picker",
                    "formatString": "DD-MM-YYYY HH:mm",
                    "readonly": true,
                    "flex": "0 0 32%"
                  },
                  {
                    "key": "acceptanceTimeOfContactDetailProcessing",
                    "title": {
                      "en": "Processing of contact details agreed at",
                      "uk": "Обробка контактних даних погоджена о"
                    },
                    "type": "datetime-picker",
                    "formatString": "DD-MM-YYYY HH:mm",
                    "readonly": true,
                    "flex": "0 0 32%"
                  }
                ]
              }
            ],
            "app": {
              "hideNameAndDescription": true
            }
          }
    - key: store/GET-EMPLOYEES-AGE
      inputSpec: |
          {
            "$schema": "http://json-schema.org/draft-07/schema#",
            "type": "object",
            "properties": {
              "employeesInfo": {
                "type": "array",
                "items": {
                  "type": "object",
                  "properties": {
                    "employeeInfo": {
                      "$ref": "#/xmDefinition/employeeInfo"
                    }
                  }
                }
              },
              "location": {
                "$ref": "#/xmDefinition/address"
              }
            },
            "required": ["employeesInfo"],
            "xmDefinition": {
              "address": {
                "type": "object",
                "properties": {
                  "address": {
                    "type": "string"
                  },
                  "geoAddress": {
                    "$ref": "#/xmDefinition/geoAddress"
                  },
                  "formattedAddress": {
                    "type": "string"
                  }
                }
              },
              "employeeInfo": {
                "$schema": "http://json-schema.org/draft-07/schema#",
                "type": "object",
                "properties": {
                  "storeName": {
                    "type": "string"
                  },
                  "info": {
                    "$ref": "#/xmDefinition/userInfo"
                  }
                }
              },
              "userInfo": {
                "$schema": "http://json-schema.org/draft-07/schema#",
                "type": "object",
                "properties": {
                  "username": {
                    "type": "string"
                  },
                  "phoneNumber": {
                    "type": "string"
                  },
                  "age": {
                    "type": "number"
                  }
                }
              }
            }
          }
      inputForm: |
          {
            "form": [
              {
                "form": [
                  {
                    "key": "storeName",
                    "title": "Enter store name",
                    "type": "rest-select",
                    "labelField": "name",
                    "valueField": "id"
                  },
                  {
                    "key": "name"
                  }
                ],
                "app": {
                  "hideNameAndDescription": true,
                  "formHeaderTitle": {
                    "en": "Add new"
                  }
                }
              },
              {
                "key": "role",
                "title": {
                  "en": "Role"
                },
                "type": "string",
                "readonly": true,
                "condition": {
                  "functionBody": "{ return $.testValues; }"
                }
              },
              {
                "type": "submit",
                "title": "Submit",
                "condition": {
                  "functionBody": "{return $.principal.hasPrivilegesInline(['EMPLOYEE.UPDATE.FIELDS']);}"
                }
              }
            ],
            "app": {
              "hideNameAndDescription": true
            }
          }
definitions:
    - key: address
      ref: test-spec/json-definitions/address.json
    - key: employeeInfo
      value: |-
          {
            "$schema": "http://json-schema.org/draft-07/schema#",
            "type": "object",
            "properties": {
              "storeName": {
                "type": "string"
              },
              "info": {
                "$ref": "#/xmDefinition/userInfo"
              }
            }
          }
    - key: userInfo
      value: |-
          {
            "$schema": "http://json-schema.org/draft-07/schema#",
            "type": "object",
            "properties": {
              "username": {
                "type": "string"
              },
              "phoneNumber": {
                "type": "string"
              },
              "age": {
                "type": "number"
              }
            }
          }
forms:
    - key: itemsByStoreForm
      value: |-
          {
            "key": "storeName",
            "title": "Enter store name",
            "type": "rest-select",
            "labelField": "name",
            "valueField": "id"
          }
    - key: itemByCategoryForm
      value: |-
          {
            "form": [
              {
                "$ref": "#/xmForm/itemsByStoreForm"
              },
              {
                "key": "name"
              }
            ],
            "app": {
              "hideNameAndDescription": true,
              "formHeaderTitle": {
                "en": "Add new"
              }
            }
          }
    - key: datesForm
      ref: test-spec/json-forms/datesForm.json
